pydra/engine/specs.py

```diff
@@ -40,26 +41,44 @@ def collect_additional_outputs(self, input_spec, inputs, output_dir):
     def hash(self):
         """Compute a basic hash for any given set of fields."""
         from .helpers import hash_function
-        from .helpers_file import hash_file
 
         inp_dict = {}
         for field in attr_fields(self):
             if field.name in ["_graph_checksums", "bindings"] or field.metadata.get(
                 "output_file_template"
             ):
                 continue
-            if field.type == File and "container_path" not in field.metadata:
-                inp_dict[field.name] = hash_file(getattr(self, field.name))
-            elif isinstance(getattr(self, field.name), tuple):
-                inp_dict[field.name] = list(getattr(self, field.name))
-            else:
-                inp_dict[field.name] = getattr(self, field.name)
+            inp_dict[field.name] = self.hash_value(
+                value=getattr(self, field.name), tp=field.type, metadata=field.metadata
+            )
         inp_hash = hash_function(inp_dict)
         if hasattr(self, "_graph_checksums"):
             return hash_function((inp_hash, self._graph_checksums))
         else:
             return inp_hash
 
+    def hash_value(self, value, tp=None, metadata=None):
```

Comment:
Is this intended to permit subclasses overriding the `hash_value` method? If not, I think it would be simpler for this to be a function.

Reply:
ok, will move it